dialer.go

```diff
@@ -255,6 +268,34 @@ func (i *instrumentedConn) Close() error {
 	return nil
 }
 
+func (d *Dialer) reportOpenConns(ctx context.Context) {
+	// TODO: make this metrics reporting interval configurable.
+	t := time.NewTicker(30 * time.Second)
+	defer t.Stop()
+	for range t.C {
```

Comment:
would this make more sense to make for loop unconditional and change the default case to t.stop? Otherwise this goroutine will run for a bit of time after those channels after actually closed.

Reply:
I've dropped the loop in favor or reporting when they metrics happen. This way we let OpenCensus deal with intervals.